core/src/main/scala/org/apache/spark/HttpFileServer.scala

```diff
@@ -20,6 +20,7 @@ package org.apache.spark
 import java.io.File
 
 import com.google.common.io.Files
+import org.apache.commons.httpclient.util.URIUtil
```

Comment:
This only accidentally works since the library is a transitive dependency of core. I'm hesitant to make it a direct dependency and make core yet bigger, but in theory it already is (indirectly). It's not otherwise used in Spark though.

I know we've been around a lot on this; I had though using the JDK URI class directly was simplest rather than drag in a dependency. If anything I was questioning wrapping it up in a method, but, that's not that bad.

Any other opinions?